### ISSUE TITLE
[BUG] Cannot Combine `example` and `code` in `Prefer` Header Using `prism mock`

### ISSUE THREAD

**Issue by loganknecht** [stoplightio/prism#1693]
## Describe the bug

Your documentation states
> Remember: you can combine code, example and dynamic parameters. By default, when the code parameter is not given, Prism will always try to fetch an example using the exampleKey from the 200 HTTP responses only. You'll need to use the code parameter alongside the example one for any specific example using a different HTTP status code than 200.

https://meta.stoplight.io/docs/prism/docs/getting-started/03-cli.md#dynamic-response

However it does not provide any examples of how this is done.

And this feature does not appear to work correctly

## To Reproduce

```yaml
---
openapi: "3.0.2"

servers:
  - url: http://localhost:8080
    description: Local deployment from your computer

paths:
  /api/users/register:
    post:
      summary: "Register New User"
      description: "Register a new user"
      operationId: "registerNewUser"
      tags: ["Users"]
      requestBody:
        required: true
        content:
          application/json:
            schema:
              type: "object"
              properties:
                username:
                  type: "string"
                  minLength: 2
                  maxLength: 5
                  example: "John"
                email:
                  type: "string"
                  minLength: 4
                  maxLength: 100
                  example: "john@example.com"
                password:
                  type: "string"
                  minLength: 8
                  maxLength: 50
                  example: "abc123456"
              required:
                - username
                - email
                - password
      responses:
        "201":
          description: Successful registration of a new user
        "422":
          description: Failed to register new user
          content:
            application/json:
              schema:
                type: "object"
                properties:
                  code:
                    type: "string"
                    description: "Internal error code received"
                  message:
                    type: "string"
                    description: "The error string version of the internal error"
              examples:
                usernameIsTooShort:
                  value:
                    message: "The username is too short"
                usernameIsTooLong:
                  value:
                    message: "The username is too long"
                usernameIsEmpty:
                  value:
                    message: "A username MUST be provided"
```

### With Both Headers Using Semi Colon

![image](https://user-images.githubusercontent.com/1308524/108437194-ae50d980-7201-11eb-9fbb-e60964af0a8e.png)

### With Both Headers Using Comma

![image](https://user-images.githubusercontent.com/1308524/108437218-bf014f80-7201-11eb-9f3e-637d97ace377.png)

## Expected behavior

I expect to be able to target `code` and `example` at the same time

## Additional context

Please update your documentation with an example for this

**Screenshots**

**Environment (remove any that are not applicable):**
Prism: 4.1.2
OS: Mac OS 10.15.7

**Comment by radzserg** [comment]
Looks like it not up to date anymore and has been fixed in [#1786](https://github.com/stoplightio/prism/pull/1786) 

```

openapi: "3.0.2"

info:
  version: 1.0.0
  title: Swagger Petstore
  license:
    name: MIT

servers:
  - url: http://localhost:8080
    description: Local deployment from your computer

paths:
  /api/users/register:
    post:
      summary: "Register New User"
      description: "Register a new user"
      operationId: "registerNewUser"
      tags: ["Users"]
      requestBody:
        required: true
        content:
          application/json:
            schema:
              type: "object"
              properties:
                username:
                  type: "string"
                  minLength: 2
                  maxLength: 5
                  example: "John"
                email:
                  type: "string"
                  minLength: 4
                  maxLength: 100
                  example: "john@example.com"
                password:
                  type: "string"
                  minLength: 8
                  maxLength: 50
                  example: "abc123456"
              required:
                - username
                - email
                - password
          multipart/form-data:
            schema:
              type: "object"
              properties:
                username:
                  type: "string"
                  minLength: 2
                  maxLength: 5
                  example: "John"
                email:
                  type: "string"
                  minLength: 4
                  maxLength: 100
                  example: "john@example.com"
                password:
                  type: "string"
                  minLength: 8
                  maxLength: 50
                  example: "abc123456"
              required:
                - username
                - email
                - password
      responses:
        "201":
          description: Successful registration of a new user
          content:
            application/json:
              schema:
                type: "object"
                properties:
                  id:
                    type: "number"
                    description: "Internal error code received"
                  name:
                    type: "string"
                    description: "The error string version of the internal error"
                examples:
                  user_john:
                    value:
                      id: 10
                      name: John
        "422":
          description: Failed to register new user
          content:
            application/json:
              schema:
                type: "object"
                properties:
                  code:
                    type: "string"
                    description: "Internal error code received"
                  message:
                    type: "string"
                    description: "The error string version of the internal error"
              examples:
                usernameIsTooShort:
                  value:
                    message: "The username is too short"
                usernameIsTooLong:
                  value:
                    message: "The username is too long"
                usernameIsEmpty:
                  value:
                    message: "A username MUST be provided"

```

Request

```
# without preferred code 
 curl --location --request POST 'http://127.0.0.1:4010/api/users/register' \
--header 'Prefer: example=usernameIsEmpty' \
--header "Content-Type: application/json" \
--data '{"username":"xyz","password":"xyz123456", "email": "john@example.com"}'
{"type":"https://stoplight.io/prism/errors#NOT_FOUND","title":"The server cannot find the requested content","status":404,"detail":"Response for contentType: application/json and exampleKey: usernameIsEmpty does not exist."}%


# with preferred code 
➜  cli git:(master) curl --location --request POST 'http://127.0.0.1:4010/api/users/register' \
--header 'Prefer: example=usernameIsEmpty,code=422' \
--header "Content-Type: application/json" \
--data '{"username":"xyz","password":"xyz123456", "email": "john@example.com"}'
{"message":"A username MUST be provided"}%

```

**Comment by chohmann** [comment]
Closing this issue as it was resolved with https://github.com/stoplightio/prism/issues/1786 and released in version 4.2.
